### ISSUE TITLE
Failed to format warning

### ISSUE THREAD

**Issue by bernardd** [jeremyjh/dialyxir#389]
### Environment

* Elixir 1.10.2
* Erlang 22.2.6
* dialyxer: 1.0.0

### Current behavior
Received the following output:

```
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a:agenda_id | :circle_id | :email | :id | :name | :organization_id | :role | :scope | {:app_session_id | :author_id | :circle_id | :completion_and_assigned_to | :conference_id | :entity | :person_id | :role_id, :agenda_id | :circle_id | :completion_and_assigned_to | :entity | :inserted_at | :person_id | :role_id} :: :ok\ndef a() do\n  :ok\nend\n"


Legacy warning:
src/dynamodb.erl:506: Guard test 'agenda_id' | 'circle_id' | 'email' | 'id' | 'name' | 'organization_id' | 'role' | 'scope' | {'app_session_id' | 'author_id' | 'circle_id' | 'completion_and_assigned_to' | 'conference_id' | 'entity' | 'person_id' | 'role_id','agenda_id' | 'circle_id' | 'completion_and_assigned_to' | 'entity' | 'inserted_at' | 'person_id' | 'role_id'} =:= Index::'true' can never succeed
```

**Comment by jeremyjh** [comment]
Can you give us the output of mix dialyzer --format raw please?

**Comment by bernardd** [comment]
On further examination I think the issue here (and with a bunch of similar stuff I'm seeing) is that I'm running it on `.erl` rather than `.ex` files. If, as I suspect, you've never really implemented support for that set of output (and I totally get why you wouldn't) then I'm happy to just use `--raw` when I run it and close this.

**Comment by bernardd** [comment]
If, on the other hand, you have the inclination to support output from `.erl` files, I'm happy to give you the raw output and any further issues I find - just let me know.

**Comment by jeremyjh** [comment]
`dialyxir` doesn't run on source files at all, it looks at the compiled BEAM files in the ebin directory. I'd say its not commonly used on .erl files because Rebar has a similar module and that is what most Erlang projects would use. It is possible you will find more uncovered clauses this way simply because you may produce some compiled code that Elixir never produces but I'd say its all fair game if we can reproduce it.

**Comment by bernardd** [comment]
Yep, sorry, I meant beam files that came from `.erl` files :) Some of the conventions (variables starting with upper-case, etc) trip up some of the pattern matching. Apologies for not getting back to this yet - busy with work etc etc - I do still fully intend to get you some examples.
